packages/platforms/accton/x86-64/as5915-18x/modules/builds/x86-64-accton-as5915-18x-fan.c

```diff
@@ -0,0 +1,496 @@
+/*
+ * A hwmon driver for the Accton as5915_18x fan
+ *
+ * Copyright (C) 2016 Accton Technology Corporation.
+ * Brandon Chuang <brandon_chuang@accton.com.tw>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+
+#include <linux/module.h>
+#include <linux/jiffies.h>
+#include <linux/i2c.h>
+#include <linux/hwmon.h>
+#include <linux/hwmon-sysfs.h>
+#include <linux/err.h>
+#include <linux/mutex.h>
+#include <linux/sysfs.h>
+#include <linux/slab.h>
+#include <linux/dmi.h>
+#include <linux/platform_device.h>
+
+#define DRVNAME "as5915_18x_fan"
+#define MAX_FAN_SPEED_RPM	22000
+
+static struct as5915_18x_fan_data *as5915_18x_fan_update_device(struct device *dev);
+static ssize_t fan_show_value(struct device *dev, struct device_attribute *da, char *buf);
+static ssize_t set_duty_cycle(struct device *dev, struct device_attribute *da,
+            const char *buf, size_t count);
+static ssize_t show_version(struct device *dev, struct device_attribute *da,
+             char *buf);
+extern int as5915_18x_cpld_read(unsigned short cpld_addr, u8 reg);
+extern int as5915_18x_cpld_write(unsigned short cpld_addr, u8 reg, u8 value);
+
+/* fan related data, the index should match sysfs_fan_attributes
+ */
+static const u8 fan_reg[] = {
+    0x22, /* fan 1 present/fault status */
+    0x32, /* fan 2 present/fault status */
+    0x42, /* fan 3 present/fault status */
+    0x52, /* fan 4 present/fault status */
+	0x20, /* fan 1 speed(rpm) */
+	0x30, /* fan 2 speed(rpm) */
+	0x40, /* fan 3 speed(rpm) */
+	0x50, /* fan 4 speed(rpm) */
+	0x21, /* fan 1 pwm */
+	0x31, /* fan 2 pwm */
+	0x41, /* fan 3 pwm */
+	0x51, /* fan 4 pwm */
```

Comment:
Please format it consistently, and use the Linux kernel coding style for the whole file.